### DISCUSSION TITLE
Subpixel Granularity for `InputElement.OnPointerMoved`

### DISCUSSION THREAD

**Discussion by djrecipe** [AvaloniaUI/Avalonia#16676] [Q&A] [answered]
Hello, I am overriding this event in a `OpenGlControlBase` class, and I have found the following:

- `OnPointerMoved` fires even when there is only tiny pointer movements (expected behavior)
- However, `PointerEventArgs.GetPosition()` only yields whole numbers (unexpected behavior)

The result is that you get constant redundant firing of `OnPointerMoved` with no way to know the subpixel position. I am using pointer movement for fine camera control in a 3D environment, so it's important for me to know the exact "subpixel" position of the pointer. Is this possible?

### Example
A series of event invocations might look like this:
**OnPointerMoved** x=35, y=106
**OnPointerMoved** x=35, y=106 // redundant invocation
**OnPointerMoved** x=35, y=106 // redundant invocation
**OnPointerMoved** x=35, y=106 // redundant invocation
**OnPointerMoved** x=36, y=106 // finally rolled over to 36

... Not only do the redundant invocations (with no actual detectable movement) cause an issue for me, but the sudden jump to the next whole number results in choppy movement within my 3D environment.

**Comment by maxkatz6** [comment]
I wander if we should add something like:
```
public PixelPoint RawPosition { get; }
```

Our pointer API is based on UWP, and UWP also has this property https://learn.microsoft.com/en-us/uwp/api/windows.ui.input.pointerpoint.rawposition?view=winrt-26100
**Except**, it's still in DIPs.

**Reply by maxkatz6** [reply]
Nevermind, see https://github.com/AvaloniaUI/Avalonia/discussions/16676#discussioncomment-10321444

**Comment by maxkatz6** [comment, marked as answer]
`Point.X` and `Point.Y` are double properties. And shouldn't be rounded to integer numbers.
What platform are you running it on?

**Comment by djrecipe** [comment]
Alas, it seems it was user error. Thank you for the prompt response. Your answer was sufficient to motivate me to dig deeper and determine the root cause **in my own code**.

I am forcing the cursor back to center using `CGWarpMouseCursorPosition()` and it seems that truncates the values. See https://github.com/panda3d/panda3d/issues/1312#issuecomment-1140035651

I will close this for now and continue my work. Thanks again for the input.

**Reply by danipen** [reply]
@djrecipe, are you integrating Panda3D with Avalonia? If so, I'm interested. Are there any public repos or resources I could check out? Thanks!!
